### PR TITLE
Add GitHub Action to do CI builds on this repository

### PR DESCRIPTION
**Pull request by Krenair**
Right now we rely on Concourse's (not great) PR resource to do this in
autom8/internal-apps.
I have not tested this and am not sure how.

### CODE REVIEW

File: .github/workflows/rubyandnode.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Ruby and Node
+
+on:
```
Comment (chrisfarms) [inline .github/workflows/rubyandnode.yaml:3]: I _think_ we probably want to use `pull_request` here...

```
on:
  pull_request:
    types: [ opened, synchronize, reopened ]
```

this should:

* avoid it trying to run the Action for every single commit in the PR (just do the HEAD)
* possibly avoid running after merging `master` (no point, it's too late - not 100% sure if `push` counts here, it's not clear)

but LGTM

**Review by chrisfarms — DISMISSED** [review]

**Comment by kevindew** [comment]
A couple of minor things, you don't need to specify the ruby-version as it'll pick it up from .ruby-version file, you shouldn't need to do `gem install bundler` as bundler is packaged with Ruby 2.6 up and you may want to configure the gem install to be in parallel and cache the gems. We have some GOV.UK docs on our conventions for GitHub Actions you might find helpful: https://docs.publishing.service.gov.uk/manual/test-and-build-a-project-with-github-actions.html

Also, from a quick glance I do wonder if you need nodejs installed? maybe I missed something that uses it but I do wonder if you'll find it still works fine without node.

**Comment by Krenair** [comment]
The .ruby-version file says 2.5.3 but Concourse was using 2.6 in https://github.com/alphagov/tech-ops/blob/master/reliability-engineering/pipelines/internal-apps.yml#L237-L260
As for node:
```
alex@GDS8957-alex:~/gds-way (add-github-actions)$ docker run --entrypoint /bin/bash --mount type=bind,source="$(pwd)",target=/app -it govsvc/aws-ruby:2.6.1
root@bc6ca2770da2:/# cd app
root@bc6ca2770da2:/app# bundle install --without development
[...]
root@bc6ca2770da2:/app# bundle exec middleman build
bundler: failed to load command: middleman (/usr/local/bundle/bin/middleman)
ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes.
  /usr/local/bundle/gems/execjs-2.7.0/lib/execjs/runtimes.rb:58:in `autodetect'
  /usr/local/bundle/gems/execjs-2.7.0/lib/execjs.rb:5:in `<module:ExecJS>'
  /usr/local/bundle/gems/execjs-2.7.0/lib/execjs.rb:4:in `<top (required)>'
  /usr/local/bundle/gems/coffee-script-2.4.1/lib/coffee_script.rb:1:in `require'
  /usr/local/bundle/gems/coffee-script-2.4.1/lib/coffee_script.rb:1:in `<top (required)>'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/renderers/coffee_script.rb:2:in `require'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/renderers/coffee_script.rb:2:in `<top (required)>'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/core_extensions/rendering.rb:11:in `require'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/core_extensions/rendering.rb:11:in `block in <top (required)>'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:91:in `load'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:127:in `block in load_settings'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:125:in `each'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:125:in `load_settings'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extension_manager.rb:12:in `initialize'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/application.rb:263:in `new'
  /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/application.rb:263:in `initialize'
  /usr/local/bundle/gems/middleman-cli-4.3.11/bin/middleman:49:in `new'
  /usr/local/bundle/gems/middleman-cli-4.3.11/bin/middleman:49:in `<top (required)>'
  /usr/local/bundle/bin/middleman:23:in `load'
  /usr/local/bundle/bin/middleman:23:in `<top (required)>'
root@bc6ca2770da2:/app# 
```

**Comment by kevindew** [comment]
> The .ruby-version file says 2.5.3 but Concourse was using 2.6 in https://github.com/alphagov/tech-ops/blob/master/reliability-engineering/pipelines/internal-apps.yml#L237-L260
> As for node:
> 
> ```
> alex@GDS8957-alex:~/gds-way (add-github-actions)$ docker run --entrypoint /bin/bash --mount type=bind,source="$(pwd)",target=/app -it govsvc/aws-ruby:2.6.1
> root@bc6ca2770da2:/# cd app
> root@bc6ca2770da2:/app# bundle install --without development
> [...]
> root@bc6ca2770da2:/app# bundle exec middleman build
> bundler: failed to load command: middleman (/usr/local/bundle/bin/middleman)
> ExecJS::RuntimeUnavailable: Could not find a JavaScript runtime. See https://github.com/rails/execjs for a list of available runtimes.
>   /usr/local/bundle/gems/execjs-2.7.0/lib/execjs/runtimes.rb:58:in `autodetect'
>   /usr/local/bundle/gems/execjs-2.7.0/lib/execjs.rb:5:in `<module:ExecJS>'
>   /usr/local/bundle/gems/execjs-2.7.0/lib/execjs.rb:4:in `<top (required)>'
>   /usr/local/bundle/gems/coffee-script-2.4.1/lib/coffee_script.rb:1:in `require'
>   /usr/local/bundle/gems/coffee-script-2.4.1/lib/coffee_script.rb:1:in `<top (required)>'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/renderers/coffee_script.rb:2:in `require'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/renderers/coffee_script.rb:2:in `<top (required)>'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/core_extensions/rendering.rb:11:in `require'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/core_extensions/rendering.rb:11:in `block in <top (required)>'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:91:in `load'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:127:in `block in load_settings'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:125:in `each'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extensions.rb:125:in `load_settings'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/extension_manager.rb:12:in `initialize'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/application.rb:263:in `new'
>   /usr/local/bundle/gems/middleman-core-4.3.11/lib/middleman-core/application.rb:263:in `initialize'
>   /usr/local/bundle/gems/middleman-cli-4.3.11/bin/middleman:49:in `new'
>   /usr/local/bundle/gems/middleman-cli-4.3.11/bin/middleman:49:in `<top (required)>'
>   /usr/local/bundle/bin/middleman:23:in `load'
>   /usr/local/bundle/bin/middleman:23:in `<top (required)>'
> root@bc6ca2770da2:/app# 
> ```

Ah darn, sorry about the Node suggestion then.

**Review by chrisfarms — APPROVED** [review]

**Comment by Krenair** [comment]
No worries :)

**Pull request merged**